packages/ketcher-polymer-editor-react/src/components/monomerLibrary/MonomerLibrary.tsx

```diff
@@ -13,67 +13,169 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  ***************************************************************************/
-
-import { MonomerItem } from './monomerLibraryItem/MonomerItem'
+import { MonomerItemType } from './monomerLibraryItem'
 import { Tabs } from 'components/shared/Tabs'
-import { Switcher } from 'components/rna/Switcher'
 import styled from '@emotion/styled'
+import { RnaMonomerSection } from 'components/rna/MonomerSection'
+import { MonomerGroup } from './monomerLibraryGroup'
+
+export type Group = {
+  groupItems: Array<MonomerItemType>
+  groupTitle?: string
+}
 
-interface MonomerListPropTypes {
-  list: Array<Record<string, string>>
+interface MonomerListProps {
+  list: Array<Group>
+  onItemClick: (item) => void
 }
 
 const monomerItemHandleClick = () => {
   console.log('callback for monomer item')
 }
 
-const MonomerLibrary = () => {
-  const MonomerList = (props: MonomerListPropTypes) => {
-    return props.list.map((monomer, key) => {
-      return (
-        <MonomerItem
-          key={key}
-          item={monomer}
-          onClick={monomerItemHandleClick}
-        />
-      )
-    })
-  }
+const MonomerList = ({ list, onItemClick }: MonomerListProps) => {
+  return (
+    <>
+      {list.map(({ groupTitle, groupItems }, key) => {
+        return (
+          <MonomerGroup
+            key={key}
+            title={groupTitle}
+            items={groupItems}
+            onItemClick={onItemClick}
+          />
+        )
+      })}
+    </>
+  )
+}
 
+const MonomerLibrary = () => {
   const tabs = [
```

Comment:
We shouldn't declare both `tabs` and `MonomerLibraryContainer` right within `MonomerLibrary` component, this can have performance issues. You can simply move them up outside the component.

Also, maybe move `tabs` content to a separate file as some sort of a constant?